Lib/test/test_compile.py

```diff
@@ -2055,6 +2055,39 @@ def test_lambda_return_position(self):
                         self.assertGreaterEqual(end_col, start_col)
                         self.assertLessEqual(end_col, code_end)
 
+    def test_return_in_with_positions(self):
+        # See gh-98442
+        def f():
+            with xyz:
+                1
+                2
+                3
+                4
+                return R
+
+        # All instructions should have locations on a single line
+        for instr in dis.get_instructions(f):
+            start_line, end_line, _, _ = instr.positions
+            self.assertEqual(start_line, end_line)
+
+        # Expect three load None instructions for the no-exception __exit__ call,
+        # and one RETURN_VALUE.
+        # They should all have the locations of the context manager ('xyz').
```

Comment:
Why doesn't the RETURN_VALUE have the location of the return statement?

Reply:
The calculation of the value has that location. The return itself occurs after the __exit__, so it gets the location of the __exit__ (the last instruction executed before the return).